### PR TITLE
(PC-16585) feat(search): use algolia query suggestions index

### PR DESCRIPTION
**Pull request by cedricls1**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-16585

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

**Review by bruno-ebstein-pass-culture — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -183,6 +183,7 @@ commands:
             echo "BATCH_API_KEY_IOS=$BATCH_API_KEY_IOS_STAGING" >> .env.staging
             echo "CONTENTFUL_ACCESS_TOKEN=$CONTENTFUL_ACCESS_TOKEN_STAGING" >> .env.staging
             echo "ALGOLIA_SEARCH_API_KEY=$ALGOLIA_SEARCH_API_KEY_STAGING" >> .env.staging
+            echo "ALGOLIA_SUGGESTIONS_API_KEY=$ALGOLIA_SEARCH_API_KEY_STAGING" >> .env.staging
```
Comment (kopax-polyconseil) [inline .circleci/config.yml:186]: rm

File: .circleci/config.yml [hunk 2]
```diff
@@ -209,6 +210,7 @@ commands:
             echo "BATCH_API_KEY_IOS=$BATCH_API_KEY_IOS_PRODUCTION" >> .env.production
             echo "CONTENTFUL_ACCESS_TOKEN=$CONTENTFUL_ACCESS_TOKEN_PRODUCTION" >> .env.production
             echo "ALGOLIA_SEARCH_API_KEY=$ALGOLIA_SEARCH_API_KEY_PRODUCTION" >> .env.production
+            echo "ALGOLIA_SUGGESTIONS_API_KEY=$ALGOLIA_SEARCH_API_KEY_PRODUCTION" >> .env.production
```
Comment (kopax-polyconseil) [inline .circleci/config.yml:213]: rm

File: src/features/search/pages/Search.tsx [hunk 3]
```diff
@@ -15,15 +15,15 @@ import { SearchHeader } from 'features/search/components/SearchHeader'
...
+import { AlgoliaSuggestionHit } from 'libs/algolia'
 import { env } from 'libs/environment'
 import { OfflinePage } from 'libs/network/OfflinePage'
 import { useNetInfo } from 'libs/network/useNetInfo'
 import { Spacer } from 'ui/theme'
 import { Form } from 'ui/web/form/Form'
 
 const searchInputID = uuidv4()
-const algoliaClient = algoliasearch(env.ALGOLIA_APPLICATION_ID, env.ALGOLIA_SEARCH_API_KEY)
+const algoliaClient = algoliasearch(env.ALGOLIA_APPLICATION_ID, env.ALGOLIA_SUGGESTIONS_API_KEY)
```
Comment (kopax-polyconseil) [inline src/features/search/pages/Search.tsx:26]: TODO: use `client.ts`

File: src/libs/environment/types.ts [hunk 4]
```diff
@@ -4,6 +4,8 @@ export interface Environment {
   ALGOLIA_OFFERS_INDEX_NAME: string
   ALGOLIA_SEARCH_API_KEY: string
   ALGOLIA_VENUES_INDEX_NAME: string
+  ALGOLIA_SUGGESTIONS_INDEX_NAME: string
+  ALGOLIA_SUGGESTIONS_API_KEY: string
```
Comment (kopax-polyconseil) [inline src/libs/environment/types.ts:8]: rm

File: src/libs/environment/__mocks__/envFixtures.ts [hunk 5]
```diff
@@ -5,6 +5,8 @@ export const env: Environment = {
   ALGOLIA_OFFERS_INDEX_NAME: 'algoliaOffersIndexName',
   ALGOLIA_SEARCH_API_KEY: 'algoliaApiKey',
   ALGOLIA_VENUES_INDEX_NAME: 'algoliaVenuesIndexName',
+  ALGOLIA_SUGGESTIONS_INDEX_NAME: 'algoliaSuggestionsIndexName',
+  ALGOLIA_SUGGESTIONS_API_KEY: 'algoliaSuggestionsApiKey',
```
Comment (kopax-polyconseil) [inline src/libs/environment/__mocks__/envFixtures.ts:9]: rm

File: .env.testing [hunk 6]
```diff
@@ -63,6 +63,8 @@ ALGOLIA_APPLICATION_ID=testingHXXTDUE7H0
 ALGOLIA_SEARCH_API_KEY=97720afb9a013cd92d656ec9ac9811a3
 ALGOLIA_OFFERS_INDEX_NAME=TESTING
 ALGOLIA_VENUES_INDEX_NAME=testing-venues
+ALGOLIA_SUGGESTIONS_INDEX_NAME=TESTING_query_suggestions
+ALGOLIA_SUGGESTIONS_API_KEY=ca032d8a5da4530ffc307de0b44d594f
```
Comment (kopax-polyconseil) [inline .env.testing:67]: rm

**Pull request merged**